File: Lab2-MAP/src/TaskContainerFactory.java
public class TaskContainerFactory implements Factory{

    private static TaskContainerFactory instance;
    private TaskContainerFactory() {
        super();
    }
    public static TaskContainerFactory getInstance() {
        if (instance == null) {
            instance = new TaskContainerFactory();
        }
        return instance;
    }

    @Override
    public Container createContainer(Strategy strategy) {
        switch(strategy){
            case FIFO:
                return new QueueContainer();
            case LIFO:
                return new StackContainer();
            default: return null;
        }
    }
}

File: Lab9-MAP/Lab9-MAP/TaskContainerFactory.cs
namespace Lab9_MAP;

using System;

public class TaskContainerFactory : IFactory
{
    public  IContainer CreateContainer(Strategy strategy)
    {
        switch (strategy)
        {
            case Strategy.FIFO:
                return new QueueContainer();
            case Strategy.LIFO:
                return new StackContainer();
            default:
                return null;
        }
    }
}
